commands/setup/setup.go

```diff
@@ -21,19 +21,30 @@ package setup
 import (
 	"fmt"
 
+	"github.com/apache/skywalking-infra-e2e/internal/components/setup"
 	"github.com/apache/skywalking-infra-e2e/internal/config"
-
 	"github.com/apache/skywalking-infra-e2e/internal/constant"
+	"github.com/apache/skywalking-infra-e2e/internal/logger"
+	"github.com/apache/skywalking-infra-e2e/internal/util"
 
 	"github.com/spf13/cobra"
-
-	"github.com/apache/skywalking-infra-e2e/internal/components/setup"
-	"github.com/apache/skywalking-infra-e2e/internal/logger"
 )
 
 var Setup = &cobra.Command{
 	Use:   "setup",
 	Short: "",
+	PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
+		err := util.CheckDockerDaemon()
+		if err != nil {
+			return err
+		}
+
+		err = config.ReadGlobalConfigFile(constant.E2EDefaultFile)
```

Comment:
The `config.ReadGlobalConfigFile(constant.E2EDefaultFile)` should in root command? Because other command such as `e2e cleanup` `e2e run` `e2e verify` ...  may also need to read the `e2e.yaml`.

Reply:
If we put `config.ReadGlobalConfigFile(constant.E2EDefaultFile)` in root command, we will not be able to verify with flags when there is no config file. 

Maybe we should define an error type, like `CfgFileNotExistError`? If the returned error is that, then just log the information.

Reply:
> If we put `config.ReadGlobalConfigFile(constant.E2EDefaultFile)` in root command, we will not be able to verify with flags when there is no config file.
> 
> Maybe we should define an error type, like `CfgFileNotExistError`? If the returned error is that, then just log the information.

This is feasible.

I'm also thinking whether we could simply change `GlobalE2EConfig.Ready` to `GlobalE2EConfig.Exist` or `GlobalE2EConfig.Error`, because if we load it from root command, it's guaranteed that it must be loaded / ready if it exists, so commands that really need the config can check this and use it, or return the error when `GlobalE2EConfig.Error` is non nil.

Reply:
I think change `GlobalE2EConfig.Ready` to `GlobalE2EConfig.Error` and check it before use it is appropriate. It can meet the needs of a variety of situations.